heron/schedulers/src/java/com/twitter/heron/scheduler/ecs/EcsScheduler.java

```
@@ -0,0 +1,183 @@
+
+  private volatile boolean isTopologyKilled = false;
+  private File tempDockerFile = null;
+  @Override
+  public void initialize(Config mConfig, Config mRuntime) {
+    this.config = Config.toClusterMode(mConfig);
+    this.runtime = mRuntime;
+
+  }
+
+  public void close() {
+
+  }
+
+
+  protected int startExecutorSyncProcess(int container) {
+    String executingInShell = new String();
+    executingInShell = getExecutorCommand(container)[0];
+    System.out.println("executing in Shell: " + executingInShell);
+    return ShellUtils.runProcess(executingInShell, null);
+  }
+
+  protected void startExecutor(final int container) {
+    LOG.info("Starting a new executor for container: " + container);
+    int shellOutput = startExecutorSyncProcess(container);
+    LOG.info("output value for the executor container: "
+        + container + String.valueOf(shellOutput));
+
+  }
+
+  private String[] getExecutorCommand(int container) {
+    List<Integer> freePorts = new ArrayList<>(SchedulerUtils.PORTS_REQUIRED_FOR_EXECUTOR);
+    for (int i = 0; i < SchedulerUtils.PORTS_REQUIRED_FOR_EXECUTOR; i++) {
+      //freePorts.add(SysUtils.getFreePort());
+      freePorts.add(5000 + (i + (container * 10)));
+    }
+    String[] executorCmd = SchedulerUtils.executorCommand(config, runtime, container, freePorts);
+    String finalExecCommand = setClusterValues(formHeronExecCommand(executorCmd));
+    String ecsTaskProject =  EcsContext.topologyName(config) + "_" + String.valueOf(container);
+    FileOutputStream dockerFilestream = null;
+    String content = null;
+    try {
+      tempDockerFile = File.createTempFile("docker", ".yml");
+      content = getDockerFileContent(finalExecCommand, container);
+      tempDockerFile.setWritable(true);
+      dockerFilestream = new FileOutputStream(tempDockerFile);
+      IOUtils.write(content, dockerFilestream);
+      IOUtils.closeQuietly(dockerFilestream);
+    } catch (IOException  e) {
+      e.printStackTrace();
```

We should either throw an exception here, or log a warning to the logging framework, but we should not do `e.printStackTrace()`.